src/nvim/fileio.c

```diff
@@ -4333,108 +4333,100 @@ void shorten_fnames(int force)
   redraw_tabline = TRUE;
 }
 
-/*
- * add extension to file name - change path/fo.o.h to path/fo.o.h.ext
- *
- * Assumed that fname is a valid name found in the filesystem we assure that
- * the return value is a different name and ends in 'ext'.
- * "ext" MUST be at most 4 characters long if it starts with a dot, 3
- * characters otherwise.
- * Space for the returned name is allocated, must be freed later.
- * Returns NULL when out of memory.
- */
-char_u *
+/// add extension to file name - change path/fo.o.h to path/fo.o.h.ext
+///
+/// Assumed that fname is a valid name found in the filesystem we assure that
+/// the return value is a different name and ends in 'ext'.
+/// "ext" MUST be at most 4 characters long if it starts with a dot, 3
+/// characters otherwise.
+/// Space for the returned name is allocated, must be freed later.
+/// Returns NULL when out of memory.
+/// @param fname When NULL use the current directory name
+/// @param ext The extension to add to the filename, 4 chars max if prefixed 
+///            with a dot, 3 otherwise.
+/// @param prepend_dot If true, prefix the ext with a dot
+char *
 modname (
-    char_u *fname,
-    char_u *ext,
-    int prepend_dot                /* may prepend a '.' to file name */
+    char *fname,
+    char *ext,
+    bool prepend_dot  // may prepend a '.' to file name
```

Comment:
This comment isn't necessary any more, Please make this one line if possible, and it looks like `fname` and `ext` can be `const`.